shared/src/hooks/useFetchV2VaultData.ts

```diff
@@ -22,12 +22,15 @@ const useFetchV2VaultData = (): V2VaultData => {
     account: web3Account,
     library,
   } = useWeb3React();
-  const { provider: avaxProvider } = useWeb3Context(
-    isDevelopment() ? CHAINID.AVAX_FUJI : CHAINID.AVAX_MAINNET
-  );
-  const { provider: ethProvider } = useWeb3Context(
-    isDevelopment() ? CHAINID.ETH_KOVAN : CHAINID.ETH_MAINNET
-  );
+  const providers = {
+    ethereum: useWeb3Context(
+      isDevelopment() ? CHAINID.ETH_KOVAN : CHAINID.ETH_MAINNET
+    ).provider,
+    avalanche: useWeb3Context(
+      isDevelopment() ? CHAINID.AVAX_FUJI : CHAINID.AVAX_MAINNET
+    ).provider,
+    aurora: useWeb3Context(CHAINID.AURORA_MAINNET).provider,
+  };
```

Comment:
We could abstract this provider out to a more manageable API, something like:

```js
const {getProviderForNetwork} = useEVMWeb3Context()

const inferredProviderFromVault = getProviderForNetwork(getVaultNetwork(vault));

/// In shared/hooks/useEVMWeb3Context.ts

const useEVMWeb3Context = () => {
  return { getProviderFornetwork }
}
```

Reply:
I tried this but it appears to break the rules of hooks because we need to call the hook inside a map.

```Don’t call Hooks inside loops, conditions, or nested functions.```
https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Reply:
getProviderForNetwork is just a normal function returned by the `useEVMWeb3Context` hook, so it won't violate it. This is how you would do it:

```js
const {getProviderForNetwork} = useEVMWeb3Context()

VaultList.map(vault => {
 ...
 const inferredProviderFromVault = getProviderForNetwork(getVaultNetwork(vault));
})
```

Reply:
Fixed!